source/common/http/http1/codec_impl.cc

```diff
@@ -24,6 +24,17 @@ namespace Http {
 namespace Http1 {
 namespace {
 
+struct Http1ResponseCodeDetailValues {
+  const std::string TooManyHeaders = "http1.too_many_headers";
+  const std::string HeadersTooLarge = "http1.headers_too_large";
+  const std::string HttpCodecError = "http1.codec_error";
+  const std::string InvalidCharacters = "http1.invalid_characters";
+  const std::string ConnectionHeaderSanitization = "http1.connection_header_bad";
+  const std::string InvalidUrl = "http1.invalid_url";
+};
+
+using Http1ResponseCodeDetails = ConstSingleton<Http1ResponseCodeDetailValues>;
```

Comment:
These can all be `absl::string_view` instead. They're always passed to functions expecting `string_view`.

Also, honestly since there's no initialization order fiasco here involved.
Why not just declare all those literals as `constexpr auto var = "..."`?
Then you wouldn't need any allocations in the singleton and no std::string allocations either (even though many of the values here qualify for SSO)